### ISSUE TITLE
SqlServerSagaPersister ignores custom serialization settings

### ISSUE THREAD

**Issue by chilversc** [rebus-org/Rebus#400]
Using rebus 0.84 I am using `UseNewtonsoftJsonSerializer` to provide a custom serialize that has support for custom types (such as NodaTime).

This works fine for messages on the bus, but when I try and store any data from those messages in a saga the custom serializers are ignored. This leads to the data being incorrectly serialized/deserialized.

Looking at the latest source in master it seems this same problem still exists in the latest master.

I think `SqlServerSagaStorage` should have the following virtual properties to allow overriding them in derived classes:

```csharp
protected virtual string SerializeData(object sagaData)
{
    return (ISagaData)JsonConvert.DeserializeObject(value, Settings)
}

protected virtual ISagaData DeserializeData(string seriaizedSagaData)
{
    return JsonConvert.SerializeObject(sagaData, Formatting.Indented, Settings)
}
```

There are only 3 lines that then need to be changed in `SqlServerSagaStorage` to use these new functions; [L195], [L226], and [L283]

[L195]: https://github.com/rebus-org/Rebus/blob/bc36cbffe23e96cbebe11d83968978a214b340c2/Rebus/Persistence/SqlServer/SqlServerSagaStorage.cs#L195
[L226]: https://github.com/rebus-org/Rebus/blob/bc36cbffe23e96cbebe11d83968978a214b340c2/Rebus/Persistence/SqlServer/SqlServerSagaStorage.cs#L226
[L283]: https://github.com/rebus-org/Rebus/blob/bc36cbffe23e96cbebe11d83968978a214b340c2/Rebus/Persistence/SqlServer/SqlServerSagaStorage.cs#L283

**Comment by mookid8000** [comment]
It's because it's _the message serializer_ that you're configuring by calling `UseNewtonsoftJsonSerializer`.

I've never had the need to use anything but JSON.NET for serializing saga data, but I can see that it might be because my value types are usually JSON serialization friendly because I default to write my value types that way.

I did think about making the saga data serializer configurable though, but if I did that, that would be in the new Rebus. You're using the old Rebus ("Rebus 1"), and the new stuff is happening in "Rebus 2".
